libs/js-sdk/src/client.ts

```diff
@@ -79,8 +77,20 @@ export function createBrowserClient({
     debugLog('SDK connecting in background (%o)', {
         audiences,
     })
-    updateStrategyImplementation
-        .connect(state)
+
+    retry(async () => {
+        try {
+            return await updateStrategyImplementation.connect(stateStore)
```

Comment:
Can we add the debug logging in.

It might also be worth logging errors here so it's visble